### ISSUE TITLE
[BUG] Exclude '_index' from columnsOrder when 'fields' command is entered

### ISSUE THREAD

**Issue by sonamgupta21** [siglens/siglens#965]
# Description
When the fields command is entered, the _index field should not be included in the columnsOrder. This is to ensure that the output only contains the relevant fields specified by the user.

# Steps to Reproduce
Give detailed step-by-step instructions to help us easily reproduce the issue.
1. On logs search screen run a query including fields. ex `* | fields city, user_color, app_name`
2. See the backend WS response it contains `_index` in columnsOrder. 
<img width="1295" alt="image" src="https://github.com/siglens/siglens/assets/71771131/689fb268-d7f5-460a-b467-8a7da2af9f08">

# Expected Behavior
Should only include the fields user has entered and timestamp.

# Environment
- Commit Hash: [e.g. 9ed5f0a]
- OS: [e.g. MacOS 14.1.1 on apple silicon]
- Browser and Version (if applicable): [e.g. chrome 119.0.6045.159 arm64, safari 19616.2.9.11.7]

# (Optional) Additional Context
Add anything else we should know about this problem.

**Comment by ahegde3** [comment]
Can I work on this issue?

**Comment by nkunal** [comment]
Yes go ahead

**Comment by ahegde3** [comment]
@nkunal  @sonamgupta21 
I went through the code to find where we are adding **_index** to the result and found this in recordreader.go
I am not sure about the context behind this code (line 171 and line 184) but by adding some logic to not add **_index** we can fix this issue.

**<img width="885" alt="Screenshot 2024-06-08 at 9 21 03 PM" src="https://github.com/siglens/siglens/assets/20241972/5ebe58a3-037d-47c7-ab12-2c9e339897e2">** 

Let me know how should I approach this.

**Comment by nkunal** [comment]
Yes thats sounds about right about the approach. But this new logic should happen only if it is a fields query
